group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorRuntime.java

```diff
@@ -458,9 +458,9 @@ private CoordinatorContext(
         private void updateLastWrittenOffset(
             long offset
         ) {
-            if (offset <= lastWrittenOffset) {
+            if (offset < lastWrittenOffset) {
```

Comment:
Hum... I am not sure about this because the offset should always increase. This is an important invariant of the system and by relaxing this we could potentially violate it.